src/main/java/io/codis/jodis/RoundRobinJedisPool.java

```diff
@@ -199,9 +214,20 @@ private void resetPools() {
             }
         }
         this.pools = builder.build();
-        for (PooledObject pool: addr2Pool.values()) {
+        for (final PooledObject pool: addr2Pool.values()) {
             LOG.info("Remove proxy: " + pool.addr);
-            pool.pool.close();
+            new Thread(new Runnable() {
```

Comment:
Better using ScheduledThreadPool?

Reply:
Yes, ScheduledThreadPoolExecutor is better. ScheduledThreadPoolExecutor uses a priority queue to maintain tasks. The algorithm is not optimal, but it's ok for our case.